File: project/angular_project/src/app/models/company.ts
import { Address } from "./address";

export class Company {
  company_id: number;
  name: string;
  website: string;
  address: Address;

  constructor() {
    this.company_id = 0;
    this.name = '';
    this.website = '';
    this.address = new Address();
  }
}

File: src/main/java/com/example/project/entities/Company.java
package com.example.project.entities;

import java.io.Serializable;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "Company")
public class Company implements Serializable{
	@Id
	private int company_id;
	private String name;
	private String website;
    private Address address;
}